### PR TITLE
de: Translation until chapter 9.1

### PR DESCRIPTION
**Pull request by Throvn**
Like mentioned in #827 the PR currently lacks formatting and therefore is huge.
Once I have information on how to solve this, I will request review.
I would also appreciate feedback on the size of the PR, is it too big or still acceptable?

### CODE REVIEW

**Review by fniemann — APPROVED** [review]
Added some comments (before I realized how long the file is) and not all of them are relevant to your changes, so feel free to ignore the comments on lines you didn't touch.

File: po/de.po [hunk 1]
```diff
@@ -1341,43 +1327,42 @@ msgid ""
...
-"`mdbook serve` direkt präsentierst (siehe [Installationsanleitung][3]). Somit entstehen keine "
-"Störungen oder Unklarheiten bei dem Wechseln von Seiten und falls du oder die Teilnehmer "
-"Rechtschreibfehler finden können diese sofort behoben werden.\n"
-"1. Gib den Kursteilnehmern Zeit die Aufgaben selbst oder in kleinen Gruppen zu lösen. Vergiss "
-"nicht ab und an zu fragen ob jemand Hilfe benötigt. Wenn mehrere Teilnehmer das gleiche Problem "
-"haben, spreche es vor allen Teilnehmern an und zeige einen möglichen Lösungsweg auf. Z. B. "
-"könntest du zeigen wo man die relevante Information in der Standardbibliothek findet.\n"
+"Teilnehmeranzahl von 15 bis 20 Leuten. Das schafft ein gutes Klima, um Fragen zu stellen, und es ist "
+"möglich alle Fragen zeitnah zu beantworten. Stelle auch sicher, dass der Raum _Tische_ mit Stühlen für "
+"die Teilnehmer und dich hat. Ihr werdet nämlich eure Laptops benötigen. Um genau zu sein wirst du viel "
```
Comment (ronaldfw) [inline po/de.po:1356]:
```suggestion
"die Teilnehmer und dich hat. Ihr werdet nämlich eure Laptops benötigen. Um genau zu sein, wirst du viel "
```

File: po/de.po [hunk 2]
```diff
@@ -1341,43 +1327,42 @@ msgid ""
...
+"vorprogrammieren.\n"
+"1. Sei schon etwas vor dem ersten Termin da, um alles vorzubereiten. Wir schlagen vor, dass du `mdbook "
+"serve` direkt präsentierst (siehe [Installationsanleitung][3]). Somit entstehen keine Störungen oder "
+"Unklarheiten bei dem Wechseln von Seiten und falls du oder die Teilnehmer Rechtschreibfehler finden "
+"können diese sofort behoben werden.\n"
+"1. Gib den Kursteilnehmern Zeit die Aufgaben selbst oder in kleinen Gruppen zu lösen. Vergiss nicht ab "
```
Comment (ronaldfw) [inline po/de.po:1362]:
```suggestion
"1. Gib den Kursteilnehmern Zeit die Aufgaben selbst oder in kleinen Gruppen zu lösen. Vergiss nicht, ab "
```

File: po/de.po [hunk 3]
```diff
@@ -1341,43 +1327,42 @@ msgid ""
...
+"und an zu fragen ob jemand Hilfe benötigt. Wenn mehrere Teilnehmer das gleiche Problem haben, spreche es "
```
Comment (ronaldfw) [inline po/de.po:1363]:
```suggestion
"und an zu fragen, ob jemand Hilfe benötigt. Wenn mehrere Teilnehmer das gleiche Problem haben, spreche es "
```

File: po/de.po [hunk 4]
```diff
@@ -1341,43 +1327,42 @@ msgid ""
...
+"vor allen Teilnehmern an und zeige einen möglichen Lösungsweg auf. Z. B. könntest du zeigen wo man die "
```
Comment (ronaldfw) [inline po/de.po:1364]:
```suggestion
"vor allen Teilnehmern an und zeige einen möglichen Lösungsweg auf. Z. B. könntest du zeigen, wo man die "
```

File: po/de.po [hunk 5]
```diff
@@ -1783,17 +1759,17 @@ msgstr ""
 "\n"
 "   * Editionen dürfen rückwärtsinkompatible Änderungen vornehmen.\n"
 "\n"
-"   * Um zu verhindern, dass Code nicht kaputt geht, sind Editionen optional: Du wählst ihre "
-"Edition für deine Kiste (crate) in der Datei `Cargo.toml`\n"
+"   * Um zu verhindern, dass Code nicht kaputt geht, sind Editionen optional: Du wählst ihre Edition für "
```
Comment (ronaldfw) [inline po/de.po:1762]:
```suggestion
"   * Um zu verhindern, dass Code kaputt geht, sind Editionen optional: Du wählst ihre Edition für "
```

File: po/de.po [hunk 6]
```diff
@@ -1783,17 +1759,17 @@ msgstr ""
...
+"deine Kiste (crate) in der Datei `Cargo.toml`\n"
 "\n"
-"   * Um eine Aufspaltung des Ökosystems zu vermeiden, können Rust-Compiler Code der für "
-"verschiedene Editionen geschrieben wurde vermischen.\n"
+"   * Um eine Aufspaltung des Ökosystems zu vermeiden, können Rust-Compiler Code der für verschiedene "
```
Comment (ronaldfw) [inline po/de.po:1765]:
```suggestion
"   * Um eine Aufspaltung des Ökosystems zu vermeiden, können Rust-Compiler Code, der für verschiedene "
```

File: po/de.po [hunk 7]
```diff
@@ -1783,17 +1759,17 @@ msgstr ""
...
+"Editionen geschrieben wurde vermischen.\n"
```
Comment (ronaldfw) [inline po/de.po:1766]:
```suggestion
"Editionen geschrieben wurde, vermischen.\n"
```

File: po/de.po [hunk 8]
```diff
@@ -1783,17 +1759,17 @@ msgstr ""
...
 "\n"
-"   * Erwähne, dass es ziemlich selten vorkommt, dass der Compiler jemals direkt und nicht über "
-"`cargo` verwendet wird (die meisten Benutzer tun dies nie).\n"
+"   * Erwähne, dass es ziemlich selten vorkommt, dass der Compiler jemals direkt und nicht über `cargo` "
```
Comment (ronaldfw) [inline po/de.po:1768]:
```suggestion
"   * Erwähne, dass es ziemlich selten vorkommt, dass der Compiler direkt und nicht über `cargo` "
```

**Comment by ronaldfw** [comment]
@Throvn Since you asked about the size of the PR: Yes, I think this is quite a bit too large. I think the most critical part will be to separate formatting and actual additions / changes. I believe this has been discussed with @mgeisler before, not sure what the exact conclusion was. 

Ideally, and I'm not sure if this is fully possible, we'd have two types of changes: (1) Semi-automatic (run some script to get the latest English changes; formatting), (2) new translation changes. These two types of changes should not be mixed, although we probably need some corrections in the German translation when updating the English text.

**Comment by mgeisler** [comment]
> @Throvn Since you asked about the size of the PR: Yes, I think this is quite a bit too large. I think the most critical part will be to separate formatting and actual additions / changes. I believe this has been discussed with @mgeisler before, not sure what the exact conclusion was.

The discussion is mostly in #800 where people are trying to figure out how to configure Poedit. 50+ people have touched the translations by now, so it should be possible to find a good setup from all that experience :smile:

**Comment by mgeisler** [comment]
Hi @Throvn, I've created #954 to normalize the line widths first.

When that is merged, you should incorporate the review feedback above into the PR here, then copy the resulting `de.po` file to a new checkout of the repository and create a new PR from that. What will allow us to preserve both your work and the reviews done already. The resulting change can then be merged easily since it's already checked by people.

**Comment by henrif75** [comment]
> Hi @Throvn, I've created #954 to normalize the line widths first.
> 
> When that is merged, you should incorporate the review feedback above into the PR here, then copy the resulting `de.po` file to a new checkout of the repository and create a new PR from that. What will allow us to preserve both your work and the reviews done already. The resulting change can then be merged easily since it's already checked by people.

Hi @Throvn, I just merged #954.

**Pull request closed without merging**